ui/src/components/Rules/RuleGroup/AddModal/index.tsx

```diff
@@ -345,7 +345,7 @@ const AddModal = (props: AddModal) => {
         collection: {
           visibleOnRecommended: showRecommended,
           visibleOnHome: showHome,
-          deleteAfterDays: +deleteAfterRef.current.value,
+          deleteAfterDays: (arrOption === undefined || arrOption === 4) ? undefined : +deleteAfterRef.current.value,
```

Comment:
I changed an existing rule from Delete to Do nothing and got a JS error on save. The deleteAfterRef.current.value condition can be removed for this block as it will be undefined in some cases now.